flytekit/core/pod_template.py

```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from kubernetes.client.models import V1PodSpec
+
+from flytekit.exceptions import user as _user_exceptions
+
+PRIMARY_CONTAINER_DEFAULT_NAME = "primary"
+
+
+@dataclass
+class PodTemplate(object):
```

Comment:
Trying to understand some design decisions.
Why not use V1PodTemplate directly but use customized PodTemplate object here: https://k8s-python.readthedocs.io/en/stable/kubernetes.client.models.html#kubernetes.client.models.v1_pod_template.V1PodTemplate

Reply:
probably because we don’t want users to expose to that many things. We just want them to be able to customize pod_spec, label, annotations, etc, so they won’t get confused